go/ql/lib/semmle/go/frameworks/Fasthttp.qll

```diff
@@ -0,0 +1,532 @@
+/**
+ * Provides classes for working with untrusted flow sources, sinks and taint propagators
+ * from the `github.com/valyala/fasthttp` package.
+ */
+
+import go
+private import semmle.go.security.RequestForgeryCustomizations
+
+/**
+ * Provides classes for working with the [fasthttp](github.com/valyala/fasthttp) package.
+ */
+module Fasthttp {
+  /** Gets the v1 module path `github.com/valyala/fasthttp`. */
+  string v1modulePath() { result = "github.com/valyala/fasthttp" }
+
+  /** Gets the path for the root package of fasthttp. */
+  string packagePath() { result = package(v1modulePath(), "") }
+
+  /**
+   * A class when you are using Fasthttp related queries to fully supports additional steps
+   */
+  bindingset[this]
+  abstract class AdditionalStep extends string {
```

Comment:
I see, so it isn't doing anything now? I suggest thinking about which queries you would want to use it and which extra results they would get with it. Maybe make a test or two demonstrating that. This should make it more obvious whether it should be a global taint step, or just in some queries (where you should add it), or it is actually not useful.

Reply:
I don't really want to commit it if it isn't doing anything now and there isn't a concrete plan for making it do something soon. It will just get forgotten about and never used.

Reply:
In that case, can you add something in the SSRF query that uses these classes to generate additional flow steps as part of this PR? Let me know if you need help doing that.